packages/flutter/test/widgets/framework_test.dart

```diff
@@ -61,6 +61,199 @@ void main() {
+        ),
+        LayoutBuilder(
+          key: const ValueKey<int>(2),
+          builder: (BuildContext context, BoxConstraints constraints) {
+            return SizedBox(key: key);
+          },
+        ),
+      ],
+    ));
+    expect(tester.takeException(), null);
+  });
+
+  testWidgets('GlobalKey correct case 2 - can move global key from layoutbuilder to container widget', (WidgetTester tester) async {
+    final Key key = GlobalKey(debugLabel: 'correct');
+    await tester.pumpWidget(Stack(
+      textDirection: TextDirection.ltr,
+      children: <Widget>[
+        Container(
+          key: const ValueKey<int>(1),
+          child: const Placeholder(),
+        ),
+        LayoutBuilder(
+          key: const ValueKey<int>(2),
+          builder: (BuildContext context, BoxConstraints constraints) {
+            return SizedBox(key: key);
+          },
+        ),
+      ],
+    ));
+    await tester.pumpWidget(Stack(
+      textDirection: TextDirection.ltr,
+      children: <Widget>[
+        Container(
+          key: const ValueKey<int>(1),
+          child: SizedBox(key: key),
+        ),
+        LayoutBuilder(
+          key: const ValueKey<int>(2),
+          builder: (BuildContext context, BoxConstraints constraints) {
+            return const Placeholder();
+          },
+        ),
+      ],
+    ));
+    expect(tester.takeException(), null);
+  });
+
+  testWidgets('GlobalKey correct case 3 - can deal with early rebuild in layoutbuilder - move backward', (WidgetTester tester) async {
+    const Key key1 = GlobalObjectKey('Text1');
+    const Key key2 = GlobalObjectKey('Text2');
+    await tester.pumpWidget(
+      LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) => Column(
+          children: const <Widget>[
+            _Stateful(
+              child: Text(
+                'Text1',
+                textDirection: TextDirection.ltr,
+                key: key1,
+              ),
+            ),
+            _Stateful(
+              child: Text(
+                'Text2',
+                textDirection: TextDirection.ltr,
+                key: key2,
+              ),
+            ),
+          ],
+        ),
+      )
+    );
+    final _StatefulState state = tester.firstState(find.byType(_Stateful).at(1));
+    state.rebuild();
+    // Reorder the items
+
+    await tester.pumpWidget(
+      LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) => Column(
+          children: const <Widget>[
+            _Stateful(
+              child: Text(
+                'Text2',
+                textDirection: TextDirection.ltr,
+                key: key2,
+              ),
+            ),
+            _Stateful(
+              child: Text(
+                'Text1',
+                textDirection: TextDirection.ltr,
+                key: key1,
+              ),
+            ),
+          ],
+        ),
+      )
+    );
+    expect(tester.takeException(), null);
```

Comment:
`expect(tester.takeException(), null);` is redundant, if there was an exception the test would fail